tests/trio103_104.py

```diff
@@ -0,0 +1,174 @@
+import trio
+
+try:
+    pass
+except (SyntaxError, ValueError, BaseException):
+    raise
+except (SyntaxError, ValueError, trio.Cancelled) as p:  # error
+    pass
+except (SyntaxError, ValueError):
+    raise
+except trio.Cancelled as e:
+    raise e
+except trio.Cancelled as e:
+    raise  # acceptable - see https://peps.python.org/pep-0678/#example-usage
+except trio.Cancelled:  # error
+    pass
+
+# raise different exception
+except BaseException:
+    raise ValueError()  # error TRIO104
+except trio.Cancelled as e:
+    raise ValueError() from e  # error TRIO104
+except trio.Cancelled as e:
+    # see https://github.com/Zac-HD/flake8-trio/pull/8#discussion_r932737341
+    raise BaseException() from e  # error TRIO104
+
+# if
+except BaseException as e:  # error
+    if True:
+        raise e
+    elif True:
+        pass
+    else:
+        raise e
+except BaseException:  # error
+    if True:
+        raise
+except BaseException:  # safe
+    if True:
+        raise
+    elif True:
+        raise
+    else:
+        raise
+
+# loops
+except trio.Cancelled:  # error
+    while True:
+        raise
+    else:
+        raise
+except trio.Cancelled:  # error
+    for _ in "":
+        raise
+    else:
+        raise
+
+# nested try
+# in theory safe if the try, and all excepts raises - and there's a bare except.
+# But is a very weird pattern that we don't handle.
+except BaseException as e:  # error
+    try:
+        raise e
+    except ValueError:
+        raise e
+    except:
+        raise e  # error?
+except BaseException:  # safe
+    try:
+        pass
+    finally:
+        raise
+# check that nested non-critical exceptions are ignored
+except BaseException:
+    try:
+        pass
+    except ValueError:
+        pass  # safe
+    raise
+# check that name isn't lost
+except trio.Cancelled as e:
+    try:
+        pass
+    except BaseException as f:
+        raise f
+    raise e
+# don't bypass raise by raising from nested except
+except trio.Cancelled as e:
+    try:
+        pass
+    except ValueError as g:
+        raise g  # error
+    except BaseException as h:
+        raise h  # error ?
+    raise e
```

Comment:
Need help thinking on what is going on here. Is it always wrong to do `except <critical>` while inside a `except <critical>`?

Reply:
No, nesting like that can make sense depending on the contents of the inner `try:` block.

Reply:
so: nested `raise` are not okay, unless the nested except is a handler for Cancelled/BaseException?

But hmm, this will bypass the Cancelled:
```python
try:
  ...
except trio.Cancelled:
  try:
    print(1/0)
  except:
    raise
```

Reply:
I propose we just ignore this for now - get the simple cases done in this PR, and open an issue for any follow-up which we can do (much) later.

Reply:
sure - would just like an executive decision on whether to be silent about nested critical `except`s vs put up a warning and expect users to `#noqa`. Could add a third message "TRIO107 Warning: nested BaseException/trio.Cancelled" (or increment current 105&106 so they're adjacent - maybe good to not reserve error codes for WIP features in the future).
Putting that in would just be a few lines in `visit_ExceptHandler` so no big complexity hit.

If you just don't care / prefer status quo, you can just merge it as is - the PR should be done otherwise.